server/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -70,12 +71,39 @@ default Repository create(RepositoryMetadata metadata, Function<String, Reposito
     RepositoryMetadata getMetadata();
 
     /**
-     * Reads snapshot description from repository.
+     * Reads snapshot descriptions from the repository.
      *
-     * @param snapshotId  snapshot id
-     * @return information about snapshot
+     * @param context get-snapshot-info-context
      */
-    SnapshotInfo getSnapshotInfo(SnapshotId snapshotId);
+    void getSnapshotInfo(GetSnapshotInfoContext context);
+
+    /**
+     * Reads a single snapshot description from the repository
+     *
+     * @param snapshotId snapshot id to read description for
+     * @param listener   listener to resolve with snapshot description
```

Comment:
A note on threading (& assertion) would be good here too.

Reply:
++ added